BigQuery/tests/System/RegionalizationTest.php

```diff
@@ -0,0 +1,207 @@
+use Google\Cloud\Core\Exception\ServiceException;
+
+/**
+ * @group bigquery
+ * @group bigquery-regionalization
+ */
+class RegionalizationTest extends BigQueryTestCase
+{
+    const REGION_ASIA = 'asia-northeast1';
+    const REGION_US = 'US';
+    const QUERY_TEMPLATE = 'SELECT 1 FROM `%s.%s`';
+
+    private static $datasetAsia;
+    private static $tableAsia;
+
+    public static function setUpBeforeClass()
+    {
+        parent::setUpBeforeClass();
+        self::$datasetAsia = self::createDataset(
+            self::$client,
+            uniqid(self::TESTING_PREFIX),
+            ['location' => self::REGION_ASIA]
+        );
+        self::$tableAsia = self::createTable(
+            self::$datasetAsia,
+            uniqid(self::TESTING_PREFIX)
+        );
+    }
+
+    /**
+     * @dataProvider locationDataProvider
+     */
+    public function testRunQuery($location)
+    {
+        $caught = false;
+        $query = self::$client->query(
+            sprintf(
+                self::QUERY_TEMPLATE,
+                self::$datasetAsia->id(),
+                self::$tableAsia->id()
+            )
+        )->location($location);
+
+        try {
+            self::$client->runQuery($query);
+        } catch (NotFoundException $e) {
+            $caught = true;
+        }
+
+        $this->assertExceptionShouldBeThrown($location, $caught);
+    }
+
+    /**
+     * @dataProvider locationDataProvider
+     */
+    public function testRunCopyJob($location)
+    {
+        $caught = false;
+        $copyTable = self::$dataset->table(self::TESTING_PREFIX);
+        $copy = self::$tableAsia->copy($copyTable)
+            ->location($location);
+
+        try {
+            self::$tableAsia->startJob($copy);
+            self::$deletionQueue->add($copyTable);
+        } catch (NotFoundException $e) {
+            $caught = true;
+        } catch (ServiceException $e) {
+            // Swallow this, as the important bit is whether or not
+            // we get a 404.
+        }
+
+        $this->assertExceptionShouldBeThrown($location, $caught);
+    }
+
+    /**
+     * @dataProvider locationDataProvider
+     */
+    public function testRunExtractJob($location)
+    {
+        $caught = false;
+        $extract = self::$tableAsia->extract(
+            self::$bucket->object(self::TESTING_PREFIX)
+        )->location($location);
+
+        try {
+            self::$tableAsia->startJob($extract);
+        } catch (NotFoundException $e) {
+            $caught = true;
+        }
+
+        $this->assertExceptionShouldBeThrown($location, $caught);
+    }
+
+    /**
+     * @dataProvider locationDataProvider
+     */
+    public function testRunLoadJob($location)
+    {
+        $caught = false;
+        $load = self::$tableAsia->load(
+            file_get_contents(__DIR__ . '/data/table-data.json')
+        )
+            ->sourceFormat('NEWLINE_DELIMITED_JSON')
+            ->location($location);
```

Comment:
Could we get the table resource again to see that it has additional rows (that is, that the load succeeded)?